include/exadg/incompressible_navier_stokes/spatial_discretization/spatial_operator_base.cpp

```diff
@@ -523,9 +524,17 @@ SpatialOperatorBase<dim, Number>::initialize_operators(std::string const & dof_i
 
   data.unsteady_problem = unsteady_problem_has_to_be_solved();
   if(param.temporal_discretization == TemporalDiscretization::BDFDualSplittingScheme)
+  {
     data.convective_problem = false;
+  }
   else
-    data.convective_problem = param.nonlinear_problem_has_to_be_solved();
+  {
+    data.convective_problem =
+      param.convective_problem() &&
+      (param.solver_type == SolverType::Steady ||
+       (param.solver_type == SolverType::Unsteady &&
+        (param.treatment_of_convective_term == TreatmentOfConvectiveTerm::Implicit)));
```

Comment:
how about combining these three lines of code to a function `Parameters::convective_term_is_solved_implicitly()`? These lines of code are very hard to read right now. The result would be

```c++
data.convective_problem = param.convective_problem() and param.convective_term_is_solved_implicitly();
```
This would be quite intuitive: 1. Do we have a convective term, and 2. is this term solved implicitly in terms of the algebraic system of equations?

Reply:
Wouldn't it be better to rename `param.convective_problem()` to `param.convective_term()` in this case?

Reply:
Wouldn't it be better to rename param.convective_problem() to param.convective_term() in this case?